addons/udes_stock/models/abstract_stock_model.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AbstractStockModel(models.AbstractModel):
+    _name = 'abstract.stock.model'
+
+    # Whether the model can create or not
+    CREATE = False
+    # The default domain to search with a str
+    STR_DOMAIN = ['name']
+
+    def get_or_create(self, identifier, create=False):
+        """ Get or create in model, for create also need CREATE to be True
+        :args:
+            - identifier: str or int
+                The identifier to search with
+        :kwargs:
+            - create: Boolean
+                If true, and CREATE is true, then a new object is created
+        :returns:
+            The model queried
+        """
+        model = self._name
+        model_type = model.split('.')[-1]
+        # Get the domain
+        if isinstance(identifier, str):
+            domain = ['|'] if len(self.STR_DOMAIN) > 1 else []
+            domain += [(field, '=', identifier) for field in self.STR_DOMAIN]
```

Comment:
If we have more than two does still work ?  Consider using `expression.OR()`  importing it from `from odoo.osv import expression`.